### DISCUSSION TITLE
Single file configuration question

### DISCUSSION THREAD

**Discussion by Ams-Laser** [fra589/grbl-Mega-5X#287] [Q&A] [answered]
Hello,
can some one please giveme an example on how the sinle file configuration option in config.h should look like
please
thanks

**Comment by fra589** [comment]
Hello @Ams-Laser,  
  
What don't you understand in this text?
```
/* ---------------------------------------------------------------------------------------
   OEM Single File Configuration Option
   Instructions: Paste the cpu_map and default setting definitions below without an enclosing
   #ifdef. Comment out the CPU_MAP_xxx and DEFAULT_xxx defines at the top of this file, and
   the compiler will ignore the contents of defaults.h and cpu_map.h and use the definitions
   below.
*/

// Paste CPU_MAP definitions here.

// Paste default settings definitions here.

```
  
@++;  
Gauthier.

**Reply by Ams-Laser** [reply]
Thanks for your response,
should i copy the full cpumap.h file with my settings and paste in there?
Thanks 
Ed

**Comment by fra589** [comment, marked as answer]
HI @Ams-Laser,  
  
Yes you have to copy all the file from line 30 (after the line `#ifdef CPU_MAP_2560_RAMPS_BOARD // (Arduino Mega 2560) with Ramps 1.4 Board`) to line 612 (before the line `#endif // CPU_MAP_2560_RAMPS_BOARD`).  
  
Don't forget to comment the line 43 (`#define CPU_MAP_2560_RAMPS_BOARD`) at the beginning of the config.h file.  
  
However, I don't see why you want to use a single configuration file...
The file will become huge and it is much easier to find a parameter with the logical breakdown as it is:
- config.h: general configuration of Grbl,
- cpu_map.h: pin mapping for different CPU types or alternative pin layouts,
- default.h: default Grbl settings ($0 to $999).

@++;  
Gauthier.
